### ISSUE TITLE
Large file compare causing display issues

### ISSUE THREAD

**Issue by quattj** [pnedev/comparePlus#387]
I am comparing two large files (MID files converted to text) in Windows 10.
The files compare successfully, with the status bar displaying
74236 Diff Lines: 46970 Added, 24627 Removed, 1033 Moved, 583 Changed. 41551 Match
file stats 1: length: 3,738,265  lines: 67,794 
file stats 2: length: 5,128,377  lines: 90,137

However, once the compare is finished and I try to either do a search or scroll the windows, the vertical scrollbar in the longer file window starts twitching all over the place and eventually disappears, as do the displayed contents of the file.  The contents of the shorter file also disappear after about the halfway mark.

Cancelling the compare restores the correct view of both files.

Notepad++ has been updated to 8.6.5 and ComparePlus has been updated to 1.2.0

**Comment by pnedev** [comment]
Thanks for the feedback.
Unfortunately I cannot reproduce that behavior and because of that I cannot analyze the problem.
Compare plugins are not generally meant for handling big files (or files that differ a lot) because they are not natively handling the diff marking and synchronization.
If your files are not private and you could send them I can take a look and try to figure out what might be causing the problem. Otherwise I cannot do anything about that. In that case you could try using the old Compare plugin v2.0.2 to see if the problem appears again.
Do you use ComparePlus with the default plugin settings?

**Comment by pnedev** [comment]
@quattj ,
Could you tell me what are your ComparePlus plugin settings when you observed the issue?

**Comment by quattj** [comment]
In the drop down menu, only "Detect Moves" and "Show Status Bar" are checked.

![image](https://github.com/pnedev/comparePlus/assets/63088660/cba6023b-f46e-46f8-8aaa-a1a7606014f0)
[largetxt.zip](https://github.com/pnedev/comparePlus/files/15461646/largetxt.zip)

I've included the files.  It may display fine at first, but scrolling to the end or sometimes trying to jump to other spots causes it to glitch out.

I did work around it by comparing smaller areas of each file.

**Comment by pnedev** [comment]
@quattj ,
Just as an update -
I have released ComparePlus v3.0.0 (https://github.com/pnedev/comparePlus/releases/tag/cp_3.0.0) - it includes reworked algorithms that have improved accuracy and speed so your files now compare almost instantly.
But the issue you are facing is related to the files scroll synchronization after comparing and this is something that is coming from the underlying Notepad++ text and graphics handling component (Scintilla).
I will try to think of a way to circumvent this problem but it might appear impossible at the end.
I'll write again.

**Comment by quattj** [comment]
Thanks!  I have been using the updated version and can confirm that even on very large files, the comparison is now taking a few seconds at most.  Much appreciated!

I have not experienced any scroll sync issues as of late either, running in Notepad++ 8.9.3.

**Comment by pnedev** [comment]
Thank you for the feedback.

**Comment by quattj** [comment]
A followup:
I just did a compare on selections from two large files, roughly 6000 lines.  The selection in the first file is near the start, while the second file is near the end.  All lines have at least one character different.
Once the compare was complete, the first file display went blank.
As before, I can work around it by copying the selections in to new files before comparison.

**Comment by pnedev** [comment]
@quattj ,
Have you enabled the "Show only compared selections" in the "Diffs Visual Filters..." dialog? Try that if it is not enabled.
Could you share the files? It is not strictly necessary, it just will be good to try it on my side.
Thanks for the follow-up 👍

**Comment by quattj** [comment]
I haven't tried that option yet.  I'll give it a shot if the issue crops up again.  Unfortunately, I waited too long to go back and follow up on this, so I no longer remember which files I was comparing.  I'll open a new issue if needed, and be sure to include the files.  Thanks!

**Comment by quattj** [comment]
I found another set of files that cause a display issue. If I turn on the "Show only compared selections" then everything displays normally.  If the option is not on, Notepad++ becomes very laggy and the display for the shorter of the two files becomes blank.

I didn't expect to find more files so soon, so I am reopening this issue rather than starting a new one.

It happens if I either compare the entirety of both files, or the 8 nearly identical "Track" sections, which is most of the short file and the second half of the long file.

I also discovered that clearing the compare afterwards with the "Show only compared selections" option enabled, the standard file view does not correctly revert to displaying the full original files.  They are still only displaying the sections I had selected for comparison.

[03 Ragnarok Canyon tsuu.zip](https://github.com/user-attachments/files/31744317/03.Ragnarok.Canyon.tsuu.zip)

**Comment by quattj** [comment]
Screenshot of the non-reverted view after cancelling the compare

<img width="1919" height="197" alt="Image" src="https://github.com/user-attachments/assets/6a91ff01-40aa-4375-a365-1746a3312bd1" />

**Comment by pnedev** [comment]
Hi @quattj ,

OK, thanks for the example files.
I'll check the issue when I find some spare time but that might not happen very soon.

BR
